### ISSUE TITLE
Excessive battery drain 

### ISSUE THREAD

**Issue by Alias3333** [signalapp/Signal-Android#12341]
- [ ] I have searched open and closed issues for duplicates
- [ ] I am submitting a bug report for existing functionality that does not work as intended
- [ ] I have read https://github.com/signalapp/Signal-Android/wiki/Submitting-useful-bug-reports
- [ ] This isn't a feature request or a discussion topic

----------------------------------------

### Bug description
Describe here the issue that you are experiencing.

### Steps to reproduce
- using hyphens as bullet points
- list the steps
- that reproduce the bug

**Actual result:** Describe here what happens after you run the steps above (i.e. the buggy behaviour)
**Expected result:** Describe here what should happen after you run the steps above (i.e. what would be the correct behaviour)

### Device info

**Device:** Manufacturer Model XVI
**Android version:** 0.0.0
**Signal version:** 0.0.0

### Link to debug log

https://debuglogs.org/android/5.42.7/c7b1be0c78f85a3b2a937a657ee325af04675bebf433f621f5e74e6439a97e7e

**Comment by alex-signal** [comment]
Please fill out the sheet properly, and check if there are other issues related to this one first.

**Comment by caveman1973** [comment]
I noticed Signal showing on my battery most use list.
![IMG_20220731_100138](https://user-images.githubusercontent.com/1135956/182005252-0993ea23-9cba-4b39-a06f-3763d6b76379.jpg)

I'm using GrapheneOS in a Pixel 5. Battery optimization is disabled since I'm not using Google Services for push notifications.

Signal is version 5.44.3. previous two versions showed the same problem.

**Comment by a13xl** [comment]
Same issue on Pixel 6a with GrapheneOS without Google play services.

- Download Signal from Aurora Store (or Homepage)
- Start Signal
- Give Access to Contacts, Sensor and Network
- Try Battery use optimazed and unrestricted
- Close App
- Wait for a while (e.g. 1h)
- Check Battery use in Settings: Signal is running all the time and burn down battery (20% of Battery use for 10 min usage, whole day, is normal)


Pixel 6a
Android 12
Build-Num. SD2A.220601.004.2022080500
Signal 5.44.4

Debug:
https://debuglogs.org/android/5.44.4/c31749b0bb1ff1e8a99be8fe1cad3dc8cfbc7055f04d791e4dcb7eaf16759702

**Comment by furai** [comment]
Exactly the same issue with LineageOS 19.1 on OnePlus9T without google services. I guess the custom signal service is maintaining connection at all times instead of polling in intervals which could probably lower the battery drain. I don't need to get every message instantly and I'd be happy to have a setting allowing me to check for new messages every 5 or so minutes.

**Comment by caveman1973** [comment]
I had a Pixel 3 and for the past years I've never seen Signal in the top list of apps that used more battery. For me it's recent. +3 months I think. I don't want to install an old version to test it

**Comment by furai** [comment]
Once I get my hands on a free device to fiddle with I could probably debug it. But it won't happen for another month and I have barely any experience with Android development. Anything in particular what to look for? Any tips other than link above?

**Comment by ahaverty** [comment]
I hope it's appropriate to drop some info here on what seems to be the same issue recently for myself. I can't say I've seen these warnings from signal over the last year, but seem to get them regularly for the last few days.

Signal v5.48.2
OnePlus 7T Pro
OxygenOS v11.0.9.1.HD01BA (Android 11)

![Screenshot_20220828-123801~2.jpg](https://user-images.githubusercontent.com/2920947/187072430-435e43a5-3f04-4a22-ae19-067e87886eed.jpg)

![Screenshot_20220828-123812~2.jpg](https://user-images.githubusercontent.com/2920947/187072438-11e5f490-f35b-409c-84b8-caa78b2c903d.jpg)

![Screenshot_20220828-123818.jpg](https://user-images.githubusercontent.com/2920947/187072441-064fb34d-2805-41a8-a571-07e56665e213.jpg)

**Comment by furai** [comment]
I use `Don't optimise` option per recommendation from signal cause otherwise you can stop receiving messages altogether.

**Comment by b0she** [comment]
I had to delete Signal on my phone, it killed my battery. Samsung A52s 5G
![image](https://user-images.githubusercontent.com/103332525/187922897-f2ea2767-8d7c-4d9b-9411-2f57135beb24.png)

**Comment by FuzzyQuills** [comment]
Have the exact same issue here, Pixel 4 running GrapheneOS based on Android 13. Weirdly though I only started seeing the awful battery life after the Android 13 update so I'm not 100% sure it's entirely Signal's fault.
![Screenshot_20220905-142432](https://user-images.githubusercontent.com/8356982/188360255-42e84bdc-58be-4b6d-a8d8-f2032c854c66.png)
Incidentally I'll be giving battery historian a go, I've done a tiny bit of android dev and may be able to isolate what could be causing this. (My guess is the websocket code's going haywire due to changes in Android 13)

**Comment by Toasterbirb** [comment]
I'm also experiencing this battery draining issue with Molly-FOSS, a Signal fork without google play services and I think it uses websockets for notifications.

My battery drain is at this level
![image](https://user-images.githubusercontent.com/48382422/189815189-8db28f5f-fc63-4a64-aff7-8b90ed5ab8bf.png)

I have also posted an issue about it to the respective fork repo https://github.com/mollyim/mollyim-android/issues/126 and apparently this issue is an upstream issue, so I'm also posting about it here

The issue contains some more information and a debug log

**Comment by drkhsh** [comment]
Same issue on a Pixel 6 with GrapheneOS without Google Play Services. Seems to be worse on mobile data than on WiFi.

**Comment by FedFrog** [comment]
Same issue with Pixel 7 Pro stock, so with Android 13. VIdeocalling is borderline impossible (20% in 10 mins, super hot phone) and generally Signal takes a lot of battery even for normal chatting

**Comment by Toasterbirb** [comment]
It has been *a bit* better for me. Signal is still topping the charts, but not with as outrageous numbers as before. My phones battery lasts at least for 1-2 days, so its at acceptable levels I'd say.


> Same issue with Pixel 7 Pro stock, so with Android 13. VIdeocalling is borderline impossible (20% in 10 mins, super hot phone) and generally Signal takes a lot of battery even for normal chatting

Can confirm. I made a group video call and my phone got extremely hot during it. I think other participants in the call also had to plug their phones into a charger because it was draining so much battery. Their devices had google play services installed though, so that's most likely an entirely different issue

**Comment by starbrights** [comment]
As of me I wouldn't call battery drain excessive, but definitly more than other messangers/apps.
I am running it without playservices, but with microG instead. It seems not support it's push service, although it would be possible. I just have no choice.

**Comment by 0-5-0** [comment]
I've noticed on my Pixel 6a phone with GrapheneOS that if I launch Signal Messenger then back out of it or press home (leaving it in the "system tray" let's say) then the battery usage is quite high. If I however bring up the "system tray" and clear all apps from the "system tray" or background then the Signal battery usage goes way down and the Signal background service still seems to work fine (I get calls/messages).

This is opposite to how I've seen it work when using the Tutanota client for example. If I clear the Tutanota client from the "system tray" then battery usage goes way up and if I leave it running in the "system tray" then battery usage is way down.

After a fresh reboot for example without launching any apps then Tutanota would use lots of battery but Signal would use little battery.

After launching both apps without clearing the "system tray" then Signal battery usage would go up but Tutanota's would go down.

I would hope that there's a smart way for each app to find out if they're cached and then behave accordingly.

Has anyone experienced this behavior?

**Comment by ermirry** [comment]
I am also experiencing severe battery drain with signal in the background. I'm using a Pixel 4a with stock Android 13. My battery went from 50% to dead overnight with signal being the biggest battery user in each two hour section of the battery usage graph shown below

![image](https://user-images.githubusercontent.com/45404375/207359942-19b24edc-f428-4dac-9206-9475aa5e381e.png)

**Comment by starbrights** [comment]
I found the reason for my higher power consumption (compared to my previous installation). I use microG and this includes cloud messaging. Usually this can be used by apps, but it seems that Signal doesn't use it first. So it has to do this job on it's own. After delete and reinstall it recognises the microG service and returns to a moderate power drain.

**Comment by s7m0n** [comment]
Same/similar issue here.
Not at home so no adb or further tools at hand atm. 
Just now charging to 100% for another cycle. Any hints on what to try/test.
Currently Signal is set to use "Intelligent Control" for Battery.
Somehow it seems to stay active and not really go to idle even in background.
Device is a OnePlus 7T

![signalbatterydrain](https://user-images.githubusercontent.com/22402942/207437245-a6694e72-4675-49e5-930d-120a3b7b4b3d.jpeg)

**Comment by ermirry** [comment]
So after my comment yesterday, I changed Signals battery profile in Android 13 from Optimized to Restricted and charged the device to 100%. Waking up this morning, my overnight battery drain is waaaay better. I lost about 10% battery overnight vs the 50% lost when signal was set to Optimized.

Signal is definitely putting in over time in the background when it shouldn't be.

**Comment by cody-signal** [comment]
To those newly posting on this issue, are you using the background websocket without google push notifications or using push notifications?

Additionally, if you are able to reproduce this issue on a test device, providing an Android Bug Report is the most helpful for looking at battery drain issues. I call out test device because an Android Bug Report contains a ton of personal information we'd rather you not send us and likely you'd rather not share.

Here's the tool we like to use to look at battery usage, feel free to use it yourself and report back what you see:
https://developer.android.com/topic/performance/power/battery-historian

**Comment by ermirry** [comment]
@cody-signal since I am using stock Android 13 (not rooted), I am assuming that I am just using google's push notifications as is.

I'm going to look around for a test device that supports the latest version of Signal to see if I can gather more data.

**Comment by s7m0n** [comment]
Hi,

just to be sure on the push notification questions. If i go to App-Info on Signal and then notifications all are enabled.
Or would i need to check another place as well. Unfortunately i currently do have only my main device available.
When back home i might be able to try battery historian to look on my own bug report - but to be honest that will not be soon (currently short on free time and not back till next year).

In case of other things for testing - let me know.
Let me also say thanks for providing Signal and your efforts

**Comment by cody-signal** [comment]
The easiest way would be to look at your debuglog and in the top block there will be a line like:

`FCM           : <false|true>`

If it's false, you are in websocket mode, and if true, you are using Google's push notification system. Based on your device details I'm going to guess you are in push mode, but it's worth double checking.

**Comment by s7m0n** [comment]
Seems to be active
Line  37165: 12-16 09:21:47.509 10273 32606 28065 D IncomingMessageObserver: Network: true, Foreground: true, **FCM: tru**e, Stay open requests: [], Censored: false, Registered: true, Proxy: false, Force websocket: false

**Comment by s7m0n** [comment]
@cody-signal will try - i assume it would also contain some sensitive data?
What timeframe would be reasonable as minimum). Some other steps to do (e.g. batterystats reset?)

**Comment by cody-signal** [comment]
Our debuglog logs sent from the Settings->Help screen is already anonymized and safe to share publicly. However if you wish, you can send the link to me directly via email at cody@signal.org

**Comment by s7m0n** [comment]
Back at home i could not observe this issue on my phone any more. It might have been related to the different environment i had there but today my battery stats report a reasonable 4.77%.
Sorry that i could not provide a log yet.

**Comment by greyson-signal** [comment]
Hi there, thank you very much for taking the time to run that test and share the results!

I'm not sure exactly how Element works here, but if they have "30s sync intervals", that makes me think that they turn the radio on every 30s to check for messages. Signal doesn't do that. We keep the websocket open all the time, so there's no delay when receiving messages. So I'd agree that we'd likely use more battery.

**Comment by mndscp** [comment]
Fixed the insane battery drain on my new device.
TL;DR: Making sure Signal gets installed when Google Play Services are already available seems to have fixed it for me.

Pixel 3a, GrapheneOS, Android 12: Never had this issue :heavy_check_mark: 
Pixel 5, GrapheneOS, Android 13: Did have that issue :x: 

[A few comments up](https://github.com/signalapp/Signal-Android/issues/12341#issuecomment-1349130621) somebody said that Signal might not be using Google Play Services cloud messaging and tries to do the job on its own if it doesn't find that. I checked my Signal debug log which said `FCM: false` and I remember that I installed Google Play Services after installing Signal on that phone. I uninstalled and re-installed Signal, now it shows `FCM: true` and when I check the battery overview it shows Signal usage only for the minutes that I actively used it (like I thought it should be). In contrast, before I changed this it showed Signal to be running for hours on end and drained the phone's battery almost all by itself like so many people reported in issue threads here.

I hope this helps someone else, too.

**Comment by Wolfcoool** [comment]
This is an issue for me too. Sometimes 20-30% battery drain a day. Is this still getting worked on?

**Comment by Toasterbirb** [comment]
The battery draining problem seems to have disappeared for me. I have been suspecting that the Android 13 (LineageOS 20.0, No GAPPS or MicroG installed) update did the job. For example right now Signal (or more specifically, the Molly fork) is sitting at 1% battery usage since last recharge and its usage time shows around 2 days and 3 hours, so it has definitely been running in the background.

I'm not fully certain if it was the Android version bump that fixed the issue, but I think the battery life has been way better recently after the update. This is especially cool, since Signals battery usage isn't limited in any way. I have it set to "No limits" in the app power monitoring settings and notifications via web sockets have also worked just fine.

**Comment by furai** [comment]
> For example right now Signal (or more specifically, the Molly fork) is sitting at 1% battery usage since last recharge and its usage time shows around 2 days and 3 hours, so it has definitely been running in the background.

@Toasterbirb are you using the FOSS variation?

**Comment by Toasterbirb** [comment]
> > For example right now Signal (or more specifically, the Molly fork) is sitting at 1% battery usage since last recharge and its usage time shows around 2 days and 3 hours, so it has definitely been running in the background.
> 
> @Toasterbirb are you using the FOSS variation?

Yes, version v6.18.4-1-FOSS to be exact

**Comment by furai** [comment]
> > > For example right now Signal (or more specifically, the Molly fork) is sitting at 1% battery usage since last recharge and its usage time shows around 2 days and 3 hours, so it has definitely been running in the background.
> > 
> > 
> > @Toasterbirb are you using the FOSS variation?
> 
> Yes, version v6.18.4-1-FOSS to be exact

From their docs it seems that this version uses completely different implementation for synchronising messages so it probably doesn't suffer from the same issues that original signalapp does.

**Comment by Toasterbirb** [comment]
> > > > For example right now Signal (or more specifically, the Molly fork) is sitting at 1% battery usage since last recharge and its usage time shows around 2 days and 3 hours, so it has definitely been running in the background.
> > > 
> > > 
> > > @Toasterbirb are you using the FOSS variation?
> > 
> > 
> > Yes, version v6.18.4-1-FOSS to be exact
> 
> From their docs it seems that this version uses completely different implementation for synchronising messages so it probably doesn't suffer from the same issues that original signalapp does.

I thought they both used Websockets for their push notifications if you don't have GAPPS installed. Signal defaults to FCM, but doesn't it fall back to websockets if google play services aren't present?

I have some vague memory though about there being some effort to add in alternative notification methods into Molly like ntfy or something similar. Don't quote me on this one though, since I have no clue if it went anywhere or if its even in development anymore

**Comment by greenerk** [comment]
Adding my voice here (and to another logged issue) instead of creating any new ones.
Excessive battery drain since one of the recent updates.  Been holding off  in case the issue resolved itself.

Google Pixel 6a.  Graphene OS.  No Play Services - just the "background connection" web sockets notifications.
Installed using the APK from Signal's website.  Currently version 6.20.5 but was also happening on the previous release.

Debug logs: https://debuglogs.org/android/6.20.5/37d0a416e8d034d2d89baf8d15996f5aae2ef0f0ec20a34cce018392b9574852

**Comment by Fabian-G** [comment]
Same problem here. 50% after 18 hours (mostly) idle time. I'm on a Pixel 3 with GrapheneOS without play services.
Maybe it would be easiest to just offload this problem to other projects by implementing UnifiedPush.

In any case it would be nice if there was at least a toggle to completely disable the foreground service. I tried setting the battery usage to "restricted", which does the trick, but also has the side effect that you can't listen to voice messages in the background for long.

**Comment by Undertakerrip** [comment]
As this seems not to be clarified, I want to add, that this problem started with update to Android 12 (not necessarily google Android. /e/os in my case). Another person had no issues with Android 11 like me but high consumption started for him after updating to Android 12 (google Android) as well.

That's why I guess, that something changed on Android side that must be addressed. I used socket connection without problems on Android 11 but with problems on Android 12 (same Signal version). Recent updates on Signal or Android did not fix that. Other apps using socket based notification run without that behaviour.  

I'm wondering if Signal is not tested from development side in this regard. Does this issue not occure when testing without push?

**Comment by iasdeoupxe** [comment]
As it might have missed in between i just want to add to https://github.com/signalapp/Signal-Android/issues/12341#issuecomment-1537444552 that i noticed the same:

After updating from LineageOS 19.1 (Android 12L) to LineageOS 20.0 (Android 13) on the "original" Signal app (no Molly fork involved) the situation was drastically improved and i don't see such a huge battery drain anymore.

**Comment by codeonblocks** [comment]
Experiencing the same issue here with Molly-FOSS (fork of Signal). When I showed the Molly team my logs, they said that my app is constantly trying to rebuild a network connection. If I have a VPN that rotated an IP address every so often, could that be causing the reconnection? Or is using Orbot as a proxy also a culprit?

Debug logs: https://debuglogs.org/android/6.23.5/30d9d5ef912d9f9f9315c3e3ebac7323632292ead7d8598dddbe9ee22c679193

**Comment by codeonblocks** [comment]
> As it might have missed in between i just want to add to [#12341 (comment)](https://github.com/signalapp/Signal-Android/issues/12341#issuecomment-1537444552) that i noticed the same:
> 
> After updating from LineageOS 19.1 (Android 12L) to LineageOS 20.0 (Android 13) on the "original" Signal app (no Molly fork involved) the situation was drastically improved and i don't see such a huge battery drain anymore.

What app version are you on and how's the battery life now?

**Comment by iasdeoupxe** [comment]
App version is currently 6.24.4 and battery life is stable since my previous comment, Signal is currently not even listed on the "Top 5 Apps" in the Battery Usage statistics.

**Comment by qrkourier** [comment]
Found this issue today because Signal seems to be using a problematic amount of energy as of recently. It's the top drainer in Android battery stats and I seem to be discharged sooner than normal.

Device: Pixel 6 Pro
Android: 13 TQ3A.230605.010.A1
Signal: 6.24.4
Android battery policy: optimized

**Comment by codeonblocks** [comment]
> Found this issue today because Signal seems to be using a problematic amount of energy as of recently. It's the top drainer in Android battery stats and I seem to be discharged sooner than normal.
> 
> Device: Pixel 6 Pro Android: 13 TQ3A.230605.010.A1 Signal: 6.24.4 Android battery policy: optimized

Dont use optimized if degoogled

**Comment by codeonblocks** [comment]
> App version is currently 6.24.4 and battery life is stable since my previous comment, Signal is currently not even listed on the "Top 5 Apps" in the Battery Usage statistics.

Share logs and make sure you don't have a VPN on with Signal. If you do, make it so that it bypasses Signal.

**Comment by HyperCriSiS** [comment]
Since a while now Signal is the top 1 battery draining app. Running LOS Android 11 with MicroG.

**Comment by codeonblocks** [comment]
> Since a while now Signal is the top 1 battery draining app. Running LOS Android 11 with MicroG.

If you have a VPN now that could be causing the issue. That's what caused the issue for me because the WebSocket connection kept restarting every second without success. You can view the logs in the Signal app in the settings and paste them here and someone would be able to identify what's causing the battery drain.

**Comment by HyperCriSiS** [comment]
I think it is not different with VPN, but is definitely has an issue with VPN. Often I have to restart Signal or the Tunnel to get a connection again. Other apps do not have this issue.

**Comment by codeonblocks** [comment]
> Battery drain happening again. Haven't touched my phone for 10 hours and signal drained 81% of my battery.
> 
> Pixel 4a Calyxos 4.11.3 Android 13 Signal app version 6.28.6 (after updating to this version a week or so ago, is when this issue resurfaced) No Google services, using apk from signal website Running app unrestricted for battery permissions, as I have for months.
> 
> Is there a way we can roll back to a prior version? because this makes my phone unusable. I got home from work and my battery was at 19%. Usually the battery is in the 80s.
> 
> Here is my debug log: https://debuglogs.org/android/6.28.6/1de086643c5d399b7decd94faf2e9c23fc84e23fa90d71d31d87099bd9320ae9
> 
> Thank you

View the releases of this GitHub repo, or use Molly

**Comment by johschmitz** [comment]
I can confirm that I am also having battery draining issues with the most recent Signal versions.

![Screenshot_20230826-215106569 (1)](https://github.com/signalapp/Signal-Android/assets/771979/e77b43ec-0e45-4ab0-bc5f-0db8e8e40c3d)

**Comment by johschmitz** [comment]
Here the related debug info:
![Screenshot_20230826-220652182_1](https://github.com/signalapp/Signal-Android/assets/771979/dfa28672-21e3-47e7-aad1-4070f5f1fb7e)

**Comment by jeremiah-miller** [comment]
Same issue here, latest grapheneos, signal is the biggest battery drain https://debuglogs.org/android/6.32.5/41006dca1d5618d8af704f517431f71c84f23e83538cd723646193bf395a1f3b

**Comment by jasonmeisel** [comment]
I'm getting this issue too on my Pixel 7 Pro, Android 14. I switched to Molly, and now the battery usage for background time is essentially 0%.

**Comment by Toasterbirb** [comment]
> > I'm getting this issue too on my Pixel 7 Pro, Android 14. I switched to Molly, and now the battery usage for background time is essentially 0%.
> 
> But in Molly you have to get contacts to use that correct? Doesn't communicate with signal?

You can communicate with Signal users from Molly and the other way around

**Comment by Toasterbirb** [comment]
> > > > I'm getting this issue too on my Pixel 7 Pro, Android 14. I switched to Molly, and now the battery usage for background time is essentially 0%.
> > > 
> > > 
> > > But in Molly you have to get contacts to use that correct? Doesn't communicate with signal?
> > 
> > 
> > You can communicate with Signal users from Molly and the other way around
> 
> Regarding the battery drain, which version of Molly are you using? The playstore version, fdroid unified push version, or the other fdroid version?

I have been using the FOSS version from f-droid and haven't had battery draining problems in a long while with it. Haven't tried the unified push version yet personally.

**Comment by jasonmeisel** [comment]
> > > > I'm getting this issue too on my Pixel 7 Pro, Android 14. I switched to Molly, and now the battery usage for background time is essentially 0%.
> > > 
> > > 
> > > But in Molly you have to get contacts to use that correct? Doesn't communicate with signal?
> > 
> > 
> > You can communicate with Signal users from Molly and the other way around
> 
> Regarding the battery drain, which version of Molly are you using? The playstore version, fdroid unified push version, or the other fdroid version?

I'm using the GCM version from F-Droid. There's no play store version, it just uses Google Play Services.

**Comment by steelman** [comment]
Oddly enough Signal on my device started to drain battery after I had updated the OS two days ago. Or this is just a coincidence, but the Battery Usage Signal is at the top with 21% used and a note that it has been used for 3 minutes since the device has been fully charged 13 hours ago.

A few times before I noticed that Signal drains power when it shows these in-app popups about checking PIN or donations. When I dismissed a popup, the application went to sleep properly.

Moto G7 (river)
LineageOS 20 (Android 13, w/o Play Services)
Build: 20231125
Signal 6.40.5

**Comment by steliyan** [comment]
This battery usage looks unexpectedly high to me. Given that that Signal is running only in the background without any actual usage (check screenshot below).

<img width="417" alt="image" src="https://github.com/signalapp/Signal-Android/assets/1850292/9dfa2507-ea22-43d6-9067-1af7e4e97e41">

<details><summary>Debug Log</summary>
<p>

```
============ SYSINFO ============
Time              : 1702559546085
Manufacturer      : Google
Model             : Pixel 7
Product           : panther
Screen            : 1080x2201, xxhdpi (420), 60.00 hz
Font Scale        : 1.0
Animation Scale   : 0.0
Android           : 14, API 34 (2023120800, UQ1A.231205.015.2023120800)
ABIs              : arm64-v8a
Memory            : 143M (64.27% free, 512M max)
Memclass          : 256
MemInfo           : availMem: 3454 mb, totalMem: 7479 mb, threshold: 216 mb, lowMemory: false
OS Host           : grapheneos
RecipientId       : RecipientId::2
ACI               : ********-****-****-****-*********e80
Device ID         : 1
Censored          : false
Network Status    : [isOnVpn: false, isMetered: true]
Data Saver        : DISABLED
Play Services     : false (9)
FCM               : false
BkgRestricted     : false
Locale            : en_US
Linked Devices    : true
First Version     : 1196
Days Installed    : 326
Build Variant     : playProdRelease
Emoji Version     : 8 (xhdpi)
RenderBigEmoji    : true
DontKeepActivities: false
Server Time Offset: 1727 ms (last updated: 1702559534727)
Telecom           : false
User-Agent        : Signal-Android/6.41.3 Android/34
SlowNotifications : false
PotentiallyBattery: true
App               : Signal 6.41.3 (1365, 136502) (1222c3073837) 
Package           : org.thoughtcrime.securesms (doDiFGKPO1r)
```
</p>
</details> 

Is this usage expected for a battery optimized app with a websocket connection? I have a good network coverage, so I don't expect the connection is being re-established.

**Comment by jasonmeisel** [comment]
After using it for awhile more, Molly in fact does not fix the issue. I'm not sure what causes the drain, but I did find a hint in the debug log. When it drains the battery, it outputs this once or twice a second:

`PartDataSource: Successfully loaded completed attachment file.`

**Comment by LoveIsGrief** [comment]
Using Molly with UnifiedPush mitigates the issue somewhat as one doesn't have to open Signal that often nor keep it open all the time to wait on notifications. But having Molly open for short periods of time exposes the same problem as Signal already has: amazing battery usage.

**Comment by iasdeoupxe** [comment]
There is now #13337 which has this in the description:

> This is related to Excessive battery drain #12341, but it requires some feedback before considering it fixed.

**Comment by jayb-g** [comment]
> One thing I've noticed for sure is that if you lock your screen with Signal in foreground - then it's draining battery even faster.

I found this to be true in my case too

**Comment by FuzzyQuills** [comment]
Something interesting I'll add; how many of you are doing the excessive battery drain tests on data?

I noticed on a BlackBerry Priv that if I stuck the phone on wifi all day, the battery drain from Signal is negligible, but soon as I put it on data, (4G in this case) Signal very quickly shoots up the top of the list of apps using the most power. 

Given my area of the world has one of the poorest 4G signals (below -85dB when checking with *#*#4636#*#*) is there a chance some of you are getting the excessive battery drain due to a bug triggered by poor mobile signal?

**Comment by cautious-bread** [comment]
@FuzzyQuills You may be on to something there.

- When in Wi-fi the battery use is negligible for me also.
- When on mobile data in a strong signal/reception area, it uses a little more battery, but not as much as...
- When in a poor signal/reception area the Signal chews through battery excessively.

Maybe Signal is triggering to phone to try too hard, too often for "decent" signal during Web Socket connections?

**Comment by LoveIsGrief** [comment]
It happens to me in the middle of cities while using LTE. I'm hoping https://github.com/signalapp/Signal-Android/pull/13337 will finally get merged and released to test if that's the true cause.

**Comment by jayb-g** [comment]
@FuzzyQuills @cautious-bread poor signal strength might be making it even worse. But its happening even in case of good LTE signal and in WiFi alike.

**Comment by jayb-g** [comment]
'Without GMS' seems to be the common theme here. And from the date of the issue, looks like devs not interested enough to fix it for no-GMS users.

**Comment by jm355** [comment]
It's a bummer that  #13337 is being ignored. Is there any way to bump it so the signal devs get reminded of it?

**Comment by jayb-g** [comment]
> Is there any way to bump it so the signal devs get reminded of it?

Unfortunately, no. Not that I'm aware of except this thread. 

This issue will probably be fixed on molly first and then the common/other devs would port it to signal. IMO

**Comment by csoler** [comment]
Same issue for me on /e/OS. Besides I could not find any way to prevent Signal from starting automatically in the background, so I finally had to uninstall Signal to avoid having to charge my battery every 8 hours. From experience, what usually drains battery on android apps is endless loops without a sleep, too many external accesses to WiFi/4G, constant disk access (in a loop), etc. Even without logs, looking for these in the code would certainly reveal some things.

**Comment by LoveIsGrief** [comment]
Have you maybe tried https://molly.im ? It's a soft fork of signal with some modifications to improve battery life (among others). Molly-FOSS should have the modifications.

If you're technical, you can also install Molly-UP (unified push) and the signaling server (it's a linked device that receives notifications for new messages that you have to host). I have much less battery usage on my degoogled phone now.

It's a very good alternative if you are degoogled and technical enough.

**Comment by skicek** [comment]
Since last few weeks Signal on my Nokia x30 with android 13 now 14 (still same issue) have been draining battery like hell.

Not sure if that will be helpful to find the issue, but I notice that more less it start happening when another process called Omacp started draining battery too. Not sure if those two are related.

I'm using connected mostly at home network, and going sleep with 100% battery I can wake up at the morning with 30%... Also it is in connected mode with Windows Signal App, which I used most of the day at home. 

I'm using dual sim. It works ok if I put my phone in Battery saver mode, but it is not a solution and I'm not receiving email notification then and other information.

**Comment by jm355** [comment]
The problem is still present, so the issue should stay open

**Comment by jm355** [comment]
Signal should implement unifiedpush

**Comment by furai** [comment]
Yeah, issue is still there. I've just learnt to live with this fact and I just don't let signal run in background. Means I'm missing a lot of messages right away but I've just learnt to start up signal whenever I grab my phone. Better to do that than have really short battery life.

**Comment by jayb-g** [comment]
@Ronkn since which molly version you are seeing improvements? Are you using websockets for notifications of UP?

**Comment by johschmitz** [comment]
I also installed Molly today (v7.8.1-1) to also test the power consumption behavior while being fully reachable again (no app power management restrictions). Let's see.
Of course a backup and re-import of the chat messages was necessary.

**Comment by LoveIsGrief** [comment]
Molly with unified push moves the battery usage to ntfy (the app that triggers signal when there are notifications). The total battery usage of both apps is down to 10-15% of the battery per day from whatever it was signal had (it's been so long since moving to Molly).

Even without unified push, the mentioned fix (that still hasn't been merged by the signal team) had reduced battery usage to more normal levels even before going unified push.

**Comment by jm355** [comment]
The ideal solution is definitely for signal to enable unifiedpush

**Comment by jayb-g** [comment]
@Ronkn @LoveIsGrief can you please share what is the size of User Data for Molly under App Info page? eg for me its more than 20GB. I'm trying to understand if that may also have an effect on battery drain.

**Comment by jayb-g** [comment]
> Currently using the latest version available on fdroid.

@Ronkn molly or molly-foss?

**Comment by LoveIsGrief** [comment]
My molly only has 3.27GB usage.

**Comment by jayb-g** [comment]
@Ronkn Ok, I think that's good enough for comparison

**Comment by kushagra-xo** [comment]
This problem still persists and almost make my phone hard to use with the high idle drains. A fix would be really appreciated.

**Comment by LoveIsGrief** [comment]
@kushagra-xo do you use google services?

**Comment by kushagra-xo** [comment]
> @kushagra-xo do you use google services?

no, i do not.
device: pixel 8
os: grapheneos
android version: 14

**Comment by kushagra-xo** [comment]
> The only solution to use signal without battery drain, for now, is to use [molly unifiedpush](https://github.com/mollyim/mollyim-android-unifiedpush) version with [mollysocket](https://github.com/pcrockett/mollysocket-fly). It brings down total battery usage of molly-up + ntfy up to 5% at max. As I figure it has been years since the issue and people can't wait forever for signal to fix this.

you need to self host mollysocket, which i don't find feasible for me.

**Comment by duven87** [comment]
> > The only solution to use signal without battery drain, for now, is to use [molly unifiedpush](https://github.com/mollyim/mollyim-android-unifiedpush) version with [mollysocket](https://github.com/pcrockett/mollysocket-fly). It brings down total battery usage of molly-up + ntfy up to 5% at max. As I figure it has been years since the issue and people can't wait forever for signal to fix this.
> 
> you need to self host mollysocket, which i don't find feasible for me.

Ah! with only mollyup + ntfy.sh I won't get any difference at all?
where to get my own hosted mollysocket without having to purchase a server from a provider? no distributors?

**Comment by aventura-ms** [comment]
> > It brings down total battery usage of molly-up + ntfy up to 5% at max.
> 
> @duven87 this is as good as it gets. For now. On signal you might get up to 50% or more battery usage as reported by users.
> 
> > where to get my own hosted mollysocket without having to purchase a server from a provider? no distributors?
> 
> Fly.io used to have free quota, the answer already links to the repo mollysocket-fly, which guides on how to set it up. Even if not free, it might cost you well under $5 or you can try aws but I have no experience with it. It might also be possible to self host locally but then it would involve exposing it using [expose](https://github.com/exposesh#demo) or dynamic DNS, not sure though.

 After several days of testing I have realized that the consumption has not changed anything with LOS+bitgapps on Pixel 4a. 
But yesterday I installed Langis and it's the first time I see the app in 5th position in battery consumption and only 5%. It was always the first with 20-35%.
I think I will stay with it for now.

![image](https://github.com/user-attachments/assets/08cfa599-96f3-49e4-a45e-ce24dbf0c3bc)

**Comment by aventura-ms** [comment]
that the updates go a little slower in exchange for having a phone that will last many hours more battery, I prefer it.
It seems to be updated often anyway.

![image](https://github.com/user-attachments/assets/71fc13c4-b9f0-4e1b-8853-bbd484ae6cff)

**Comment by johschmitz** [comment]
Signal from Play Store battery usage on a current stock Pixel 8:
![Screenshot_20240918-124341](https://github.com/user-attachments/assets/a87a8423-5f31-48a5-be1f-8b8d9e6f2262)

**Comment by johschmitz** [comment]
This issue seems to still persist. I now updated to Android 15 and will see if that helps somehow.

**Comment by atharva-p** [comment]
![Screenshot_20241223-082635.png](https://github.com/user-attachments/assets/d8e439e9-ed34-4004-820f-9512265b2a3d)

Signal app consumes more battery than playing Minecraft on a remote server, all while having less screen time than the game. I've also noticed overall battery drain after getting signal

**Comment by jm355** [comment]
use molly. It has a better websockets implementation, or supports unifiedpush if you set up mollysocket. Why the signal devs haven't integrated any of that is a mystery to me, but at least molly does

**Comment by jm355** [comment]
I agree it's not a solution. For one thing, a fork is always gonna be at least a little behind upstream so security fixes will take longer to get to those users, but I get minimal drain from molly using websockets. It does actually connect differently, it removes unneeded connections whereas signal has a few that stick around. So for me, the downsides of using a fork are worth good battery life

**Comment by starbrights** [comment]
@Ronkn : Does molly responds fast enough for operating calls? So does it ring?

**Comment by LinuxOnTheDesktop** [comment]
Cf. #12341.

**Comment by donnm** [comment]
I've also had this problem recently following an update to (somewhere around) 7.30.1 (now running 7.33.2) and have been trying to find the cause since. I'm using:

- Signal on LineageOS (no microG)
- Molly on LinageOS (linked device, no microG)
- Signal-Desktop on Debian (linked device)
- signal-cli arm64 on Debian (linked device, raspberry pi)

For me this problem affected latest versions of both Signal and Molly on LineageOS.

From my testing, it looks like Signal and Molly remained active (with Allow background usage set to Optimized) much more than previous versions. My phone was warm and the battery would drain in a matter of hours.

By watching messages received from signal-cli in daemon mode, I could see a flood of unusual envelopes containing  "Received a sync message" from both Signal and Molly at intervals varying roughly 1 to 4 seconds (see below, device 1 is Signal Android, device 4 is Molly):

```
Envelope from: xxx yyy-yyy-yyyy (device: 1) to yyy-yyy-yyyy                                                                                                                                                                
Timestamp: 1740054565152 (2025-02-20T12:29:25.152Z)                                                                                                                                                                         
Server timestamps: received: 1740054565309 (2025-02-20T12:29:25.309Z) delivered: 1740054565310 (2025-02-20T12:29:25.310Z)                                                                                                   
Received a sync message                                                                                                                                                                                                     
                                                                                                                                                                                                                            
Envelope from: xxx yyy-yyy-yyyy (device: 1) to yyy-yyy-yyyy                                                                                                                                                                
Timestamp: 1740054568248 (2025-02-20T12:29:28.248Z)                                                                                                                                                                         
Server timestamps: received: 1740054568410 (2025-02-20T12:29:28.410Z) delivered: 1740054568412 (2025-02-20T12:29:28.412Z)                                                                                                   
Received a sync message                                                                                                                                                                                                     
                                                                                                                                                                                                                            
Envelope from: xxx yyy-yyy-yyyy (device: 4) to yyy-yyy-yyyy                                                                                                                                                                
Timestamp: 1740054570595 (2025-02-20T12:29:30.595Z)                                                                                                                                                                         
Server timestamps: received: 1740054570902 (2025-02-20T12:29:30.902Z) delivered: 1740054570904 (2025-02-20T12:29:30.904Z)                                                                                                   
Received a sync message                                                                                                                                                                                                     
                                                                                                              
Envelope from: xxx yyy-yyy-yyyy (device: 4) to yyy-yyy-yyyy                                                                                                                                                                
Timestamp: 1740054572741 (2025-02-20T12:29:32.741Z)                                                                                                                                                                         
Server timestamps: received: 1740054573050 (2025-02-20T12:29:33.050Z) delivered: 1740054573052 (2025-02-20T12:29:33.052Z)                                                                                                   
Received a sync message                                                                                                                                                                                                     
                                                                                                                                                                                                                            
Envelope from: xxx yyy-yyy-yyyy (device: 4) to yyy-yyy-yyyy                                                                                                                                                                
Timestamp: 1740054575081 (2025-02-20T12:29:35.081Z)                                                                                                                                                                         
Server timestamps: received: 1740054575391 (2025-02-20T12:29:35.391Z) delivered: 1740054575392 (2025-02-20T12:29:35.392Z)                                                                                                   
Received a sync message                 
```

 Looking at the Signal app debug logs, I noticed messages with similar intervals from StorageSyncJob:

```
02-20 12:20:58.714  6064  6122 I StorageSyncJob: Our version: 445255.1, their version: 445256.2
02-20 12:20:58.714  6064  6122 I StorageSyncJob: [Remote Sync] Newer manifest version found!
02-20 12:20:58.792  6064  6122 I StorageSyncJob: [Remote Sync] Pre-Merge ID Difference :: remoteOnly: 1, localOnly: 0, hasTypeMismatches: false
02-20 12:20:58.792  6064  6122 I StorageSyncJob: [Remote Sync] Retrieving records for key difference.
02-20 12:20:59.917  6064  6122 I StorageSyncJob: [Remote Sync] Remote-Only :: Contacts: 1, GV1: 0, GV2: 0, Account: 0, DLists: 0, call links: 0
02-20 12:20:59.943  6064  6122 I StorageSyncJob: [Remote Sync] Unknowns :: 0 inserts, 0 deletes
02-20 12:20:59.945  6064  6122 I StorageSyncJob: [Remote Sync] Saved new manifest. Now at version: 445256.2
02-20 12:20:59.947  6064  6122 I StorageSyncJob: We are up-to-date with the remote storage state.
02-20 12:20:59.988  6064  6122 I StorageSyncJob: ID Difference :: remoteOnly: 2, localOnly: 1, hasTypeMismatches: false
02-20 12:20:59.988  6064  6122 I StorageSyncJob: We have something to write remotely.
02-20 12:20:59.988  6064  6122 I StorageSyncJob: WriteOperationResult :: ManifestVersion: 445257, Total Keys: 250, Inserts: 1, Deletes: 2
02-20 12:21:00.585  6064  6122 I StorageSyncJob: Saved new manifest. Now at version: 445257.1
02-20 12:21:00.587  6064  6122 D StorageSyncJob: [StorageSync] remote-manifest: 804, remote-id-diff: 84, remote-records: 1124, remote-merge-transaction: 29, local-data-transaction: 43, remote-write: 598, known-unknowns: 1, total: 2682
02-20 12:21:02.803  6064  6121 I StorageSyncJob: Our version: 445257.1, their version: 445258.2
02-20 12:21:02.803  6064  6121 I StorageSyncJob: [Remote Sync] Newer manifest version found!
02-20 12:21:02.850  6064  6121 I StorageSyncJob: [Remote Sync] Pre-Merge ID Difference :: remoteOnly: 1, localOnly: 0, hasTypeMismatches: false
02-20 12:21:02.850  6064  6121 I StorageSyncJob: [Remote Sync] Retrieving records for key difference.
02-20 12:21:03.168  6064  6121 I StorageSyncJob: [Remote Sync] Remote-Only :: Contacts: 1, GV1: 0, GV2: 0, Account: 0, DLists: 0, call links: 0
02-20 12:21:03.178  6064  6121 I StorageSyncJob: [Remote Sync] Unknowns :: 0 inserts, 0 deletes
02-20 12:21:03.179  6064  6121 I StorageSyncJob: [Remote Sync] Saved new manifest. Now at version: 445258.2
02-20 12:21:03.179  6064  6121 I StorageSyncJob: We are up-to-date with the remote storage state.
02-20 12:21:03.219  6064  6121 I StorageSyncJob: ID Difference :: remoteOnly: 2, localOnly: 1, hasTypeMismatches: false
02-20 12:21:03.220  6064  6121 I StorageSyncJob: We have something to write remotely.
02-20 12:21:03.220  6064  6121 I StorageSyncJob: WriteOperationResult :: ManifestVersion: 445259, Total Keys: 250, Inserts: 1, Deletes: 2
02-20 12:21:03.513  6064  6121 I StorageSyncJob: Saved new manifest. Now at version: 445259.1
02-20 12:21:03.515  6064  6121 D StorageSyncJob: [StorageSync] remote-manifest: 288, remote-id-diff: 51, remote-records: 317, remote-merge-transaction: 11, local-data-transaction: 41, remote-write: 295, known-unknowns: 1, total: 1004
02-20 12:21:05.461  6064  6118 I StorageSyncJob: Our version: 445259.1, their version: 445260.2
02-20 12:21:05.461  6064  6118 I StorageSyncJob: [Remote Sync] Newer manifest version found!
02-20 12:21:05.490  6064  6118 I StorageSyncJob: [Remote Sync] Pre-Merge ID Difference :: remoteOnly: 1, localOnly: 0, hasTypeMismatches: false
02-20 12:21:05.490  6064  6118 I StorageSyncJob: [Remote Sync] Retrieving records for key difference.
02-20 12:21:05.846  6064  6118 I StorageSyncJob: [Remote Sync] Remote-Only :: Contacts: 1, GV1: 0, GV2: 0, Account: 0, DLists: 0, call links: 0
02-20 12:21:05.856  6064  6118 I StorageSyncJob: [Remote Sync] Unknowns :: 0 inserts, 0 deletes
02-20 12:21:05.857  6064  6118 I StorageSyncJob: [Remote Sync] Saved new manifest. Now at version: 445260.2
02-20 12:21:05.857  6064  6118 I StorageSyncJob: We are up-to-date with the remote storage state.
02-20 12:21:05.892  6064  6118 I StorageSyncJob: ID Difference :: remoteOnly: 2, localOnly: 1, hasTypeMismatches: false
02-20 12:21:05.892  6064  6118 I StorageSyncJob: We have something to write remotely.
02-20 12:21:05.892  6064  6118 I StorageSyncJob: WriteOperationResult :: ManifestVersion: 445261, Total Keys: 250, Inserts: 1, Deletes: 2
02-20 12:21:06.175  6064  6118 I StorageSyncJob: Saved new manifest. Now at version: 445261.1
02-20 12:21:06.176  6064  6118 D StorageSyncJob: [StorageSync] remote-manifest: 289, remote-id-diff: 30, remote-records: 355, remote-merge-transaction: 12, local-data-transaction: 36, remote-write: 283, known-unknowns: 1, total: 1005
```

In the periods where I didn't see excessive background battery usage (there were some), these messages did not appear in the app debug log and signal-cli did not print "Received a sync message" messages to stdout (I kept an eye on this because I had to write a small Bash script to filter these from the stdout of signal-cli in order to read other messages).

I then wondered if Signal and Molly were trying to synchronise something with signal-cli, so I updated signal-cli to the latest version from https://github.com/AsamK/signal-cli/wiki/Binary-distributions. Both the Signal app debug log messages about StorageSyncJob and the signal-cli messages are now silent. Was this a bug in signal-cli?

At least in my case, this appears to have fixed the issue and reduced the battery usage to what it was before (it's still the most active app in the battery usage list). Unfortunately all of the debuglogs linked in this issue are returning 404 so I cannot check if they have the same StorageSyncJob messages.

**Comment by iasdeoupxe** [comment]
Not sure if relevant but a comment in #13337 is linking to:
- 93d18c176320ad3ba0fc9f0c317b6505ec9ff6b9
- 323697dfc902ca4bd986bcd9f3393d94ee11a0cb
- 5a0def739080d1ead117170b935638b12da5ea95

and another mentions:

> However the behavior has been added to the latest beta, 7.38.x

**Comment by mhnap** [comment]
It seems that the issue is still present in `7.40.2`.

**Comment by CristianKerr** [comment]
I started having the same issue. I am on CalyxOS and all was good just a month ago (plus minus). Now its capable of draining 50 percent of my fairly new battery in just a half day. Crazy. And crazy it is not fixed yet. And the fix seems to be so straight forward and pro-privacy that I wonder why Singal does not have unified push yet.
That said I also don't want to be just negative and want to say that I am thankful for Signal. Its genuinely cool app and I hope it will be even cooler one day.

**Comment by jm355** [comment]
Not stale, bad bot

**Comment by LinuxOnTheDesktop** [comment]
Problems - especially serious ones - going unsolved is not ideal. Having a bot close the corresponding issue for no good reason - for the reason, really, that the problem has _not_ been fixed - worsens matters.

**Comment by jm355** [comment]
Maybe we should respond to the stale bot with a link to Molly, a fork of signal android with more responsive devs who've fixed this issue https://github.com/mollyim/mollyim-android

**Comment by Tolcan** [comment]
Phone pixel 9 pro XL
OS GrapheneOS / no microG
Signal APK Ver: 7.52

Battery drain is high. Signal uses %42 of the battery

**Comment by dik23** [comment]
In what world is doing nothing the same as fixing things?

**Comment by LinuxOnTheDesktop** [comment]
> In what world is doing nothing the same as fixing things?

Well, the status 'wontfix' is not the same as fixed. Indeed, 'wontfix' implies - if taken literally - _not_ fixed. Yet, given that closed issue tend to be regarded as a non-problems, it does seem that issues ought not to be closed because 'stale'. _The problem at issue here - severe battery drain - very much seems like something that needs fixing._

**Comment by csoler** [comment]
Closing this doesn't make sense to me either. I dropped using Signal because of this, and I'm probably not alone.

**Comment by jm355** [comment]
Maybe every time the stale bot does this, we should @ the signal devs?

**Comment by wertzuz** [comment]
At least comment to unstale after the label is applied unfortunately searching on the repo it happens quite often. Maybe the des should change it to reopen after it's closed with stalebit when there is a new comment or only do the final closing with dev approval.

**Comment by jm355** [comment]
It does reopen if somebody comments after the bot (there may be a time cutoff though?), but that doesn't seem to matter. It should ping the devs when that happens though, and the devs should step in and either decide not to fix the issue, or decide it's worth fixing and stop stalebot from closing it again

**Comment by iasdeoupxe** [comment]
FWIW a few years ago a developer asked for more info and gave insights what they need:

https://github.com/signalapp/Signal-Android/issues/12341#issuecomment-1220751858

Anyone has actually send this information to them?

**Comment by jm355** [comment]
Afaik, Molly doesn't have the issue. So the solution is already there, the devs just haven't merged it

**Comment by iasdeoupxe** [comment]
Relevant mollyim/mollyim-android#126 for Molly is still open. Maybe they have merged #13337 which had the last comment of a Signal dev:

> It did. We were already planning to do an overhaul of our web socket infra, so we made sure to consider the foreground/background need aspect of it when doing that.

**Comment by jm355** [comment]
They did merge that

**Comment by iasdeoupxe** [comment]
According to a comment https://github.com/signalapp/Signal-Android/pull/13337#issuecomment-2750849304 Signal as well (could be in a different way):
https://github.com/signalapp/Signal-Android/commit/93d18c176320ad3ba0fc9f0c317b6505ec9ff6b9
https://github.com/signalapp/Signal-Android/commit/323697dfc902ca4bd986bcd9f3393d94ee11a0cb
https://github.com/signalapp/Signal-Android/commit/5a0def739080d1ead117170b935638b12da5ea95

Edit: And at least for me there is no battery drain since longer (see https://github.com/signalapp/Signal-Android/issues/12341#issuecomment-1590030081) and signal used 1% for me.

Maybe there is a different issue in Signal which isn't present in Molly and which needs to be identified first?

**Comment by iasdeoupxe** [comment]
Found two relevant comments in [mollyim/mollyim-android#126](https://github.com/mollyim/mollyim-android/issues/126#issuecomment-1903321946):

> However, for a quick test, I built molly with the https://github.com/signalapp/Signal-Android/pull/13337 patch included and it does not resolve the issue on a Fairphone 4 with /e/ OS 1.17. The issue seems unchanged: normal battery drain when on wifi, but extreme battery drain when on mobile network.

and:

[mollyim/mollyim-android#126](https://github.com/mollyim/mollyim-android/issues/126#issuecomment-1903594056):

> Yesterday we improved the patch, but haven't updated the pull-request yet. Use the beta we just released instead. It includes the latest commit https://github.com/mollyim/mollyim-android/commit/7257d571bff30e4b5759ec82b15a7993cd945866.
>
> Don't expect miracles, but you should notice less consumption.

Maybe this helps the Signal team to identify the differences.

**Comment by Neustradamus** [comment]
To follow this ticket in more this one:
- https://github.com/signalapp/Signal-Android/issues/9729

**Comment by jayb-g** [comment]
might not be stale

**Comment by jayb-g** [comment]
May not be fixed yet

**Comment by ghost** [comment]
Currently I use Molly with SunUp to get my messages through UnifiedPush. It gives great battery life. Now I was wondering if this was still needed since some patches from Molly have been applied to Signal. There also have been quite some updates since then, currently I have a Pixel 7 with android (GrapheneOS) 16. There seems to be quite some conflicting information with some not having the issue and others having the issue. So that is why I ask.

**Comment by The-Nickname** [comment]
> Currently I use Molly with SunUp to get my messages through UnifiedPush. It gives great battery life. Now I was wondering if this was still needed since some patches from Molly have been applied to Signal. There also have been quite some updates since then, currently I have a Pixel 7 with android (GrapheneOS) 16. There seems to be quite some conflicting information with some not having the issue and others having the issue. So that is why I ask.

In Molly for UnifiedPush there is still a MollySocket server needed.
If Signal would implement UnifiedPush(as alternative to GMS and Polling) than I believe this could be irrelevant.

**Comment by ghost** [comment]
Should this not be closed as a duplicate of https://github.com/signalapp/Signal-Android/issues/9729?

**Comment by lonix1** [comment]
NOT STALE... do not close!
